bokeh/cli/core.py

```diff
@@ -0,0 +1,285 @@
+import click
+
+from . import help_messages as hm
+from .utils import (get_chart_params, get_charts_mapping,
+                    get_data_series, keep_source_input_sync, get_data_from_url)
+from .. import charts as bc
+
+# Define a mapping to connect chart types supported arguments and chart classes
+CHARTS_MAP = get_charts_mapping()
+
+@click.command()
+@click.option('--input', 'input_source', default=None,help=hm.HELP_INPUT)
+@click.option('--output', default='file://cli_output.html', help=hm.HELP_OUTPUT)
+@click.option('--title', default='Bokeh CLI')
+@click.option('--plot_type', default='Line')
+@click.option('--index', default='', help=hm.HELP_INDEX)
+@click.option('--series', default='', help=hm.HELP_SERIES)
+@click.option('--palette')
+@click.option('--buffer', default='f', help=hm.HELP_BUFFER)
+@click.option('--sync_with_source', default=False)
+def cli(input_source, output, title, plot_type, series, palette,
+        index, buffer, sync_with_source):
+    """Bokeh Command Line Tool is a minimal client to access high level plotting
+    functionality provided by bokeh.charts API.
+
+    Examples:
+
+    >> python bokeh-cli.py --title "My Nice Plot" --series "High,Low,Close"
+    --plot_type "Line" --palette Reds --input sample_data/stocks_data.csv
+
+    >> cat sample_data/stocks_data.csv | python bokeh-cli.py --buffer t
+
+    >> python bokeh-cli.py --help
+    """
+    cli = CLI(
+        input_source, output, title, plot_type, series, palette, index, buffer,
+        sync_with_source
+    )
+    cli.run()
+
+
+class CLI(object):
+    """This is the Bokeh Command Line Interface class and it is in
+    charge of providing a very high level access to bokeh charts and
+    extends it with functionality.
+
+    """
+    def __init__(self, input_source, output, title, plot_type, series, palette,
+                 index, buffer, sync_with_source):
+        """Args:
+        input_source (str): path to the series data file (i.e.:
+            /source/to/my/data.csv)
+            NOTE: this can be either a path to a local file or an url
+        output (str, optional): Selects the plotting output, which
+            could either be sent to an html file or a bokeh server
+            instance. Syntax convention for this option is as follows:
+            <output_type>://<type_arg>
+
+            where:
+              - output_type: 'file' or 'server'
+              - 'file' type options: path_to_output_file
+              - 'server' type options syntax: docname[@url][@name]
+
+            Defaults to: --output file://cli_output.html
+
+            Examples:
+                --output file://cli_output.html
+                --output file:///home/someuser/bokeh_rocks/cli_output.html
+                --output server://clidemo
+
+            Default: file://cli_output.html.
+        title (str, optional): the title of your chart.
+            Default: None.
+        plot_type (str, optional): charts classes to use to consume and
```

Comment:
Yeap, you are right. When `plot_type` was added when I was using `plotting` but now it's at least confusing. Will change.